builtin/logical/database/path_roles.go

```diff
@@ -2,263 +2,596 @@ package database
+	if err != nil {
+		return nil, err
+	}
+
+	// createRole is a boolean to indicate if this is a new role creation. This is
+	// can be used later by database plugins that distinguish between creating and
+	// updating roles, and may use seperate statements depending on the context.
+	createRole := req.Operation == logical.CreateOperation
+	if role == nil {
+		role = &roleEntry{
+			StaticAccount: &staticAccount{},
 		}
+		createRole = true
+	}
 
-		return logical.ListResponse(entries), nil
+	if err := pathRoleCreateUpdateCommon(ctx, role, req.Operation, data); err != nil {
+		return logical.ErrorResponse(err.Error()), nil
 	}
-}
 
-func (b *databaseBackend) pathRoleCreateUpdate() framework.OperationFunc {
-	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-		name := data.Get("name").(string)
-		if name == "" {
-			return logical.ErrorResponse("empty role name attribute given"), nil
+	username := data.Get("username").(string)
+	if username == "" && createRole {
+		return logical.ErrorResponse("username is a required field to create a static account"), nil
+	}
+
+	if role.StaticAccount.Username != "" && role.StaticAccount.Username != username {
+		return logical.ErrorResponse("cannot update static account username"), nil
+	}
+	role.StaticAccount.Username = username
+
+	// If it's a Create operation, both username and rotation_period must be included
+	rotationPeriodSecondsRaw, ok := data.GetOk("rotation_period")
+	if !ok && createRole {
+		return logical.ErrorResponse("rotation_period is required to create static accounts"), nil
+	}
+	if ok {
+		rotationPeriodSeconds := rotationPeriodSecondsRaw.(int)
+		if rotationPeriodSeconds < queueTickSeconds {
+			// If rotation frequency is specified, and this is an update, the value
+			// must be at least that of the constant queueTickSeconds (5 seconds at
+			// time of writing), otherwise we wont be able to rotate in time
+			return logical.ErrorResponse("rotation_period must be 5 seconds or more"), nil
 		}
+		role.StaticAccount.RotationPeriod = time.Duration(rotationPeriodSeconds) * time.Second
+	}
+
+	if rotationStmtsRaw, ok := data.GetOk("rotation_statements"); ok {
+		role.Statements.Rotation = rotationStmtsRaw.([]string)
+	} else if req.Operation == logical.CreateOperation {
+		role.Statements.Rotation = data.Get("rotation_statements").([]string)
+	}
 
-		role, err := b.Role(ctx, req.Storage, data.Get("name").(string))
+	// lvr represents the roles' LastVaultRotation
+	lvr := role.StaticAccount.LastVaultRotation
+
+	// only call setStaticAccount if we're creating the role for the
+	// first time
+	switch req.Operation {
+	case logical.CreateOperation:
+		// setStaticAccount calls Storage.Put and saves the role to storage
+		resp, err := b.setStaticAccount(ctx, req.Storage, &setStaticAccountInput{
+			RoleName:   name,
+			Role:       role,
+			CreateUser: createRole,
```

Comment:
"Yes", but it's actually deferred to the plugin implementation, so it's conceivable that a plugin could not support it. 

`createRole` in this context is tracking if we created the vault role in this operation. The actual database user may or may not already exist, and what to do if it does could vary by the plugin. In this instance, the postgres plugin checks for existence of the user and if not found, uses the creation statements, otherwise it executes the rotation statements.